cpp/src/reductions/quantile.cu

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "simple.cuh"
+
+#include <cudf/detail/copy.hpp>
+#include <cudf/detail/quantiles.hpp>
+#include <cudf/detail/sorting.hpp>
+#include <cudf/reduction/detail/reduction_functions.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cudf {
+namespace reduction {
+namespace detail {
+
+std::unique_ptr<cudf::scalar> quantile(column_view const& col,
+                                       double qvalue,
+                                       cudf::interpolation interpolation,
+                                       cudf::data_type const output_type,
+                                       rmm::cuda_stream_view stream,
+                                       rmm::device_async_resource_ref mr)
+{
+  auto current_mr = cudf::get_current_device_resource_ref();
+  auto sorted_indices =
+    cudf::detail::sorted_order(table_view{{col}}, {}, {null_order::AFTER}, stream, current_mr);
+  auto valid_sorted_indices =
+    cudf::detail::split(*sorted_indices, {col.size() - col.null_count()}, stream)[0];
```

Comment:
Is there test coverage for `col.size() == col.null_count()`? (All-null inputs.) I glanced at the tests and wasn't sure if this was covered.